File: ClientSide/components/Product.js
import VLink from '../components/VLink'

export default {
    template: '<div class="product" v-bind:product="product">\
                <div class="product-image-container"><img src="./../ClientSide/assets/imgs/product-thumbnail.png"/></div>\
                <div class="product-details-container">\
                    <p>{{ product.title }}</p>\
                    <p v-html="updatedProductDescription">{{ updatedProductDescription }}</p>\
                    <p>Popularity: {{ product.popular }}</p>\
                    <v-link :href="productDetailsUrl" customClass="product-more-details-link-button">More details</v-link>\
                </div>\
                <div class="product-price-container">\
                    <p>Price: {{ product.price }} EUR</p>\
                    <p>Availability: <span class="product-availability">{{ product.availability }}</span></p>\
                </div>\
               </div>',
    components: {
        'v-link': VLink
    },
    props: {
        product: {
            type: Object,
            required: true
        }
    },
    data() {
        return {
            productDetailsUrl: "/product-details/" + this.product.id,
        }
    },
    computed: {
        updatedProductDescription: function () {
            var newDescription = this.product.description.slice(0, this.product.description.lastIndexOf(">") + 1) + " " +
                this.product.id + " " + this.product.description.slice(this.product.description.lastIndexOf(">") + 1);
            //console.log("newDescription", newDescription);
            return newDescription;
        }

    },
    methods: {

    }
}

File: Models/Product.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace ASPNET_VueJS_Example.Models
{
    public class Product
    {

        public string Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public string Image { get; set; }
        public double Price { get; set; }
        public int Popular { get; set; }
        public List<string> ListOfSpecs { get; set; }
        public string Availability { get; set; }
    }
}
